src/solution.jl

```diff
@@ -1,214 +1,332 @@
+# Here, `row_indices` corresponds to the output index of a `VectorAffineTerm`
+# Instead of `col_indices`, here we have their wrapped counterparts, `var_indices` (a vector of `MOI.VariableIndex`'s).
+function add_terms!(terms::Vector{MOI.VectorAffineTerm{T}}, matrix::SparseMatrixCSC, row_indices, var_indices) where T
+    CIs = CartesianIndices((row_indices, 1:length(var_indices)))
+    I, J, V = findnz(matrix)
+    for  k = eachindex(I, J, V)
+        term = MOI.VectorAffineTerm{T}(CIs[I[k],J[k]][1], MOI.ScalarAffineTerm{T}(V[k], var_indices[CIs[I[k],J[k]][2]]))
+        push!(terms, term)
+    end
 end
 
-function solve!(problem::Problem;
-                warmstart=false,
-                check_vexity=true,
-                verbose=true)
+function add_terms!(terms::Vector{MOI.VectorAffineTerm{T}}, vector::SparseVector, row_indices, var_indices) where T
+    CIs = CartesianIndices((row_indices, 1:length(var_indices)))
+    I, V = findnz(vector)
+    for k = eachindex(I, V)
+        term =  MOI.VectorAffineTerm{T}(CIs[I[k]][1], MOI.ScalarAffineTerm{T}(V[k], var_indices[CIs[I[k]][2]]))
+        push!(terms, term)
+    end
+end
 
-    if problem.model === nothing
-        throw(ArgumentError(
-            "The provided problem hasn't been initialized with a conic model.
-            You can resolve this by passing in a `AbstractMathProgSolver`. For example,
-            ```
-            using ECOS
-            solve!(problem, ECOSSolver())
-            ```"
-        ))
+function add_terms!(terms::Vector{MOI.VectorAffineTerm{T}}, matrix::Matrix, row_indices, var_indices) where T
+    CIs = CartesianIndices((row_indices, 1:length(var_indices)))
+    for ci = CartesianIndices(size(matrix))
+        term = MOI.VectorAffineTerm{T}(CIs[ci][1], MOI.ScalarAffineTerm{T}(matrix[ci], var_indices[CIs[ci][2]]))
+        push!(terms, term)
     end
+end
 
-    if check_vexity
-        vex = vexity(problem)
+function add_terms!(terms::Vector{MOI.VectorAffineTerm{T}}, vector::Vector, row_indices, var_indices) where T
+    CIs = CartesianIndices((row_indices, 1:length(var_indices)))
+    for k = 1:length(vector)
+        term = MOI.VectorAffineTerm{T}(CIs[k][1], MOI.ScalarAffineTerm{T}(vector[k], var_indices[CIs[k][2]]))
+        push!(terms, term)
     end
+end
 
-    c, A, b, cones, var_to_ranges, vartypes, conic_constraints, id_to_variables, conic_constr_to_constr = conic_problem(problem)
 
-    # load MPB conic problem
-    m = problem.model
-    load_problem!(m, c, A, b, cones, vartypes)
-    if warmstart
-        set_warmstart!(m, problem, length(c), var_to_ranges, id_to_variables)
-    end
-    # optimize MPB conic problem
-    MathProgBase.optimize!(m)
+# This function generates a MOI set and VectorAffineFunction corresponding to a constraint that
+# Convex represents with a `ConicConstr` object.
+# This is part of what used to be `Convex.conic_problem` (under MPB instead of MOI).
```

Comment:
Historical comments referencing how things used to be in MPB won't be too useful for someone reading this code in the future. These comments are throughout this file.

Reply:
Good point. I added those to help with review, but I probably should have put them as review comments instead of code comments.